solutions/1000-1099/1019-next-greater-node-in-linked-list.md

```diff
@@ -0,0 +1,69 @@
+# 1019 - Next Greate Node in Linked List (Medium)
+
+## Approach 1: Using stack
+
+1. Initialize an empty stack and an empty answer vector.
+2. Reverse the given linked list.
+3. Traverse the reversed linked list.
+4. For each node:
+   - While the stack is not empty and the top element is less than or equal to the current node's value, pop elements from the stack.
+   - If the stack is empty, append 0 to the answer vector.
+   - If the stack is not empty, append the top element to the answer vector.
+   - Push the current node's value onto the stack.
+5. Reverse the answer vector to obtain the correct order of next larger elements.
+6. Return the answer vector.
+
+<Tabs>
+<TabItem value="cpp" label="C++">
+<SolutionAuthor name="@yashh0903"/>
+
+```cpp
+/*
+ * Time Complexity: O(n), where n is the number of nodes in the linked list.
+ * Space Complexity: O(n), as we use a stack to store elements and an answer vector to store the result.
+ */
+
+class Solution {
```

Comment:
don't add inline comment. put the comment above that line instead. 

e.g.

```
vector<int> ans; // Stores the result
```

change it to

```
// Stores the result
vector<int> ans; 
```